server/src/test/java/org/bonitasoft/console/common/server/servlet/TenantFileUploadServletTest.java

```diff
@@ -63,8 +80,53 @@ public void checkUploadSize_should_throw_fileTooBigException_when_file_is_bigger
     @Test
     public void checkUploadSize_should_do_nothing_when_file_is_not_bigger_than_in_conf_file() throws Exception {
         when(item.getSize()).thenReturn(1000000L);
-        when(formsProperties.getAttachmentMaxSize()).thenReturn(10L); // 1Mb
+        when(formsProperties.getAttachmentMaxSize()).thenReturn(10L); // 10Mb
         fileUploadServlet.checkUploadSize(request, item);
     }
 
+    @Test
+    public void doPost_with_checkSize_should_set_413_status_code_with_empty_body_when_file_is_too_big() throws Exception {
+        doNothing().when(fileUploadServlet).defineUploadDirectoryPath(request);
+        final ServletFileUpload serviceFileUpload = mock(ServletFileUpload.class);
+        doReturn(serviceFileUpload).when(fileUploadServlet).createServletFileUpload(any(FileItemFactory.class));
+        item = mock(FileItem.class);
+        when(item.getSize()).thenReturn(20 * 1048576L);
+        when(serviceFileUpload.parseRequest(request)).thenReturn(Arrays.asList(item));
+        fileUploadServlet.uploadDirectoryPath = tempFolder.getRoot().getAbsolutePath();
+        fileUploadServlet.checkUploadedFileSize = true;
+        when(request.getMethod()).thenReturn("post");
+        when(request.getContentType()).thenReturn("multipart/");
+        final HttpServletResponse response = mock(HttpServletResponse.class);
+        final PrintWriter printer = mock(PrintWriter.class);
+        when(response.getWriter()).thenReturn(printer);
```

Comment:
Maybe some arrangement code could be factorized, simplified, looks a bit complicated.
Also it would be nice to have a clear separation (\n) between each test phases Arrange, Act, Assert